cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/RemoteTaskExecutor.java

```diff
@@ -202,4 +203,12 @@ private ByteBuffer encodeTaskRequest(RunnableTaskRequest request) throws IOExcep
 
     throw new IllegalArgumentException("Unsupported content encoding " + encoding);
   }
+
+  private void setWorkerUrl(WorkerType workerType) {
+    if (workerType == WorkerType.TASK) {
+      this.workerUrl = "/worker/run";
```

Comment:
define the string as static final at the top.

Reply:
I am not sure if this should be static, as different instances of RemoteTaskExecutor will have different url.

Reply:
It used to be static; 
`private static final String TASK_WORKER_URL = "/worker/run"`

We can now also add 
`private static final String SYSTEM_WORKER_URL = "/system/run"`

Then, you can set workerUrl to either TASK_WORKER_URL or SYSTEM_WORKER_URL

Reply:
Gotya. Done